doc/sphinx/develop/clib-extensions.md

```diff
@@ -0,0 +1,204 @@
+# Auto-Generated CLib API
+
+```{caution}
+The auto-generated CLib API is an experimental part of Cantera and may be changed
+without notice.
+```
+
+In CLib, Cantera objects are stored and referenced by integers - no pointers are passed
+to or from the calling application. Further, the contents of arrays and variables are
+copied, which separates internal C++ memory management from the application using CLib.
+
+**Example:** The CLib source generator expands the `getMolecularWeights` entry:
+
+```yaml
+- name: getMolecularWeights
+  uses: nSpecies
+```
+
```

Comment:
Perhaps it would be worth clarifying that this is part of the interface for the `ThermoPhase` class, and to provide a link to the C++ API docs for `ThermoPhase::getMolecularWeights`, to help understand that this is where the information about the method signature and docstring is coming from.